src/ffi.rs

```diff
@@ -766,6 +766,27 @@ pub extern fn quiche_conn_trace_id(
     *out_len = trace_id.len();
 }
 
+#[no_mangle]
+pub extern fn quiche_conn_source_id(
+    conn: &mut Connection, out: &mut *const u8, out_len: &mut size_t,
+) {
+    let conn_id = conn.source_id();
+    let id = conn_id.as_ref();
+    *out = id.as_ptr();
```

Comment:
Since the maximum size is already known (max cid len is 20 bytes in quic-transport spec). I think it's simpler for caller to provide 20 bytes buffer and this function copy the cid value into the provided buffer and return the actual cid len. Also who is responsible to free this pointer?

Reply:
I just thought it's a good idea to keep it the same as other functions in terms of arguments ... Also noone need to free anything as we just grab the memory which is wrapped in the id that should have the same lifetime as the connection

Reply:
@ghedo thoughts ?

Reply:
I think if possible we should try to follow what the Rust API does, so the current approach looks fine to me, but we can change later if we see copying is a better option.